prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -196,40 +196,41 @@ private void UpdateBusinessHours(Site current, CommunitySiteUpdateModel updated)
 
         private void UpdateRemoteUsers(Site current, CommunitySiteUpdateModel updated)
         {
-            // Wholesale replace the remote users
-            foreach (var remoteUser in current.RemoteUsers)
+            if (updated.RemoteUsers == null)
             {
-                foreach (var certification in remoteUser.RemoteUserCertifications)
-                {
-                    _context.Remove(certification);
-                }
-                _context.RemoteUsers.Remove(remoteUser);
+                return;
             }
 
-            if (updated?.RemoteUsers != null && updated?.RemoteUsers.Count() != 0)
+            foreach (var updateRemoteUser in updated.RemoteUsers)
             {
-                foreach (var remoteUser in updated.RemoteUsers)
-                {
-                    remoteUser.SiteId = current.Id;
-                    var remoteUserCertifications = new List<RemoteUserCertification>();
+                var existingRemoteUser = _context.RemoteUsers
+                    .FirstOrDefault(u => u.Id == updateRemoteUser.Id);
 
-                    foreach (var certification in remoteUser.RemoteUserCertifications)
+                if (existingRemoteUser == null)
+                {
+                    current.RemoteUsers.Add(updateRemoteUser);
```

Comment:
Unfortunately, we are still using models for the site update endpoint here rather than view models. Because of this, you have to be careful when taking them and using them directly like this because the front-end could have manipulated the ID (or SiteId). In this case, you should set `updatedRemoteUser.Id = 0` so EF knows its a new entity and doesn't either try to update some unrelated record or complain you are trying to add a record that already exists.

Reply:
Line 206 checks existing remote user using `Id` field already, so if it does not exist in DB, `updatedRemoteUser.Id` should be 0. If the Id has been manipulated by frontend, say setting to an existing `Id=123`, backend should do its duty to update the existing remote user but not treat it as a new remote user.